packages/custom_lint/test/cli_test.dart

```diff
@@ -32,87 +33,184 @@ Pattern progressMessage({required bool supportsAnsiEscapes}) {
+                    matches(
+                      progressMessage(
+                        supportsAnsiEscapes: ansi,
+                      ),
                     ),
+                    format == 'json'
+                        ? endsWith('{"version":1,"diagnostics":[]}\n')
+                        : endsWith('No issues found!\n'),
                   ),
-                  endsWith('No issues found!\n'),
                 ),
-              ),
-            );
-            expect(err, emitsDone);
-          },
-          currentDirectory: app,
-          supportsAnsiEscapes: supportsAnsiEscapes,
-        );
-      });
-
-      test('CLI lists warnings from all plugins and set exit code', () async {
-        final plugin =
-            createPlugin(name: 'test_lint', main: helloWordPluginSource);
-        final plugin2 = createPlugin(name: 'test_lint2', main: oyPluginSource);
-
-        final app = createLintUsage(
-          source: {
-            'lib/main.dart': 'void fn() {}',
-            'lib/another.dart': 'void fail() {}',
-          },
-          plugins: {'test_lint': plugin.uri, 'test_lint2': plugin2.uri},
-          name: 'test_app',
-        );
-
-        await runWithIOOverride(
-          (out, err) async {
-            await cli.entrypoint();
-
-            expect(err, emitsDone);
-            expect(
-              out.join(),
-              completion(
-                allOf(
-                  matches(
-                    progressMessage(
-                      supportsAnsiEscapes: supportsAnsiEscapes,
+              );
+              expect(err, emitsDone);
+            },
+            currentDirectory: app,
+            supportsAnsiEscapes: ansi,
+          );
+        });
+
+        test('CLI lists warnings from all plugins and set exit code', () async {
+          final plugin =
+              createPlugin(name: 'test_lint', main: helloWordPluginSource);
+          final plugin2 =
+              createPlugin(name: 'test_lint2', main: oyPluginSource);
+
+          final app = createLintUsage(
+            source: {
+              'lib/main.dart': 'void fn() {}',
+              'lib/another.dart': 'void fail() {}',
+            },
+            plugins: {'test_lint': plugin.uri, 'test_lint2': plugin2.uri},
+            name: 'test_app',
+          );
+
+          await runWithIOOverride(
+            (out, err) async {
+              await cli.entrypoint(['--format', format]);
+
+              final dir = IOOverrides.current!.getCurrentDirectory().path;
+              expect(err, emitsDone);
+              expect(
+                out.join(),
+                completion(
+                  allOf(
+                    matches(
+                      progressMessage(
+                        supportsAnsiEscapes: ansi,
+                      ),
                     ),
-                  ),
-                  endsWith('''
+                    format == 'json'
+                        ? endsWith('${jsonEncode({
```

Comment:
Again, construct those JSON strings out of the expectation

That expectation is way too long

Reply:
I'd suggest extracting those JSONs to a dedicated .json file

Reply:
There is some string interpolation involved, so a JSON file is not great but could do a separate part file?

Reply:
A function would be great. No need to use a part directive for that

Reply:
Done